aws/rust-runtime/aws-types/src/sdk_config.rs

```diff
@@ -822,6 +845,25 @@ impl SdkConfig {
         self.clone().into_builder()
     }
 
+    /// Set the origin of a setting.
+    ///
+    /// This is used internally to understand how to merge config structs while
+    /// respecting precedence of origins.
+    pub fn insert_origin(&mut self, setting: &'static str, origin: Origin) {
```

Comment:
Good question. I think so, but I'm not sure. For now, I'll make it so that:
- SdkConfig's Builder exposes an `insert_origin` method.
- SdkConfig itself only exposes a way to check origins.

We can add other stuff later if/when we need it.